app/containers/Forms/ProductionFields.tsx

```diff
@@ -1,127 +1,224 @@
-import React from 'react';
+import React, {useState, useCallback} from 'react';
 import {FieldProps} from 'react-jsonschema-form';
 import {createFragmentContainer, graphql} from 'react-relay';
-import {Form, Col} from 'react-bootstrap';
 import {ProductionFields_query} from 'ProductionFields_query.graphql';
+import {JSONSchema6} from 'json-schema';
+import {FormJson} from 'next-env';
 
 interface Props extends FieldProps {
   query: ProductionFields_query;
 }
 
-const ProductionFields: React.FunctionComponent<Props> = ({
+export const ProductionFieldsComponent: React.FunctionComponent<Props> = ({
   formData,
   query,
-  onChange
+  onChange,
+  registry,
+  autofocus,
+  idSchema,
+  errorSchema,
+  formContext,
+  disabled,
+  readonly,
+  schema,
+  uiSchema
 }) => {
-  const changeField = (event, key) => {
+  const {
+    FieldTemplate
+  }: {
+    FieldTemplate: React.FunctionComponent<any>;
+  } = registry as any;
+  // Not using the types definded in @types/react-jsonschema-form as they are out of date
+
+  const handleProductChange = (product: string) => {
+    const productUnits = query.allProducts.edges.find(
+      ({node}) => node.name === product
+    )?.node.units;
```

Comment:
This would be more robust matching on the product `id` not the `name`.
We don't have a graphql field to the current product?

Reply:
The value we use here will be stored in the form data (and in the database). Maybe `rowId` instead?

Reply:
Ah, yes. Good call, `rowId` would be best